src/plugins/bfetch/public/streaming/fetch_streaming.ts

```diff
@@ -26,23 +31,49 @@ export function fetchStreaming({
   method = 'POST',
   body = '',
   signal,
+  compressionDisabled$ = of(false),
 }: FetchStreamingParams) {
   const xhr = new window.XMLHttpRequest();
 
-  // Begin the request
-  xhr.open(method, url);
-  xhr.withCredentials = true;
+  const msgStream = compressionDisabled$.pipe(
+    switchMap((compressionDisabled) => {
+      // Begin the request
+      xhr.open(method, url);
+      xhr.withCredentials = true;
 
-  // Set the HTTP headers
-  Object.entries(headers).forEach(([k, v]) => xhr.setRequestHeader(k, v));
+      if (!compressionDisabled) {
+        headers['X-Chunk-Encoding'] = 'deflate';
+      }
 
-  const stream = fromStreamingXhr(xhr, signal);
+      // Set the HTTP headers
+      Object.entries(headers).forEach(([k, v]) => xhr.setRequestHeader(k, v));
 
-  // Send the payload to the server
-  xhr.send(body);
+      const stream = fromStreamingXhr(xhr, signal);
```

Comment:
converting `XMLHttpRequest --> Stream --> Observable` overcomplecates the code. Let's merge the current PR and consider switching to `DecompressionStream` after `bfetch` migration from `XMLHttpRequest` to `fetch API`.

Reply:
We don't convert to `stream` on the client. Just from `XHR` to `Observable`.
Anyway, thanks for approving. I will go ahead and merge. 👍🏻 
Any progress with `DecompressionStream`?

Reply:
Yes, I gave it a try, but as converting imperative XMLHttpRequest to Stream, with the following transformation to Observable it's PITA. That's why I suggested switching to `DecompressionStream` after migrating to `fetch` with built-in support of Streaming API.